manifests/charts/istio-control/istio-discovery/files/waypoint.yaml

```diff
@@ -35,13 +35,20 @@ spec:
             "prometheus.io/scrape" "true"
           ) | nindent 8 }}
       labels:
+        {{- $requiredLabels := .Labels }}
+        {{- $network := valueOrDefault (index .Labels `topology.istio.io/network`) .Values.global.network }}
```

Comment:
if  .Labels contains `topology.istio.io/network`, we donot need the setting below.

Why do we need add labels to waypoint, while kube-gateway we donot

Reply:
> if .Labels contains topology.istio.io/network, we donot need the setting below.

This makes sense to me since we can also set network per pod basis in injection. However I think we haven't mentioned in any places that gateway-api can be set with `topology.istio.io/network` yet.

> Why do we need add labels to waypoint, while kube-gateway we donot

The network should be set in waypoints since it impacts cross-network communication, while it doesn't directly affect the ingressgateway.

Reply:
make sense to me, so with this we donot depend on system namespace network label any more

Reply:
Will approve with test coverage